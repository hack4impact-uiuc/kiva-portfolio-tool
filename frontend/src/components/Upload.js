import React, { Component } from 'react'
import Dropzone from 'react-dropzone'
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap'
import { sendFile } from '../utils/ApiWrapper'

class Upload extends Component {
  constructor(props) {
    super(props)
    this.state = {
      files: [],
      modal: false,
      submissionStatus: ''
    }
  }

  onDrop(files) {
    this.setState({
      files
    })
  }

  toggle = () => {
    this.setState({
      modal: !this.state.modal,
      submissionStatus: 'File uploaded - your submission is being processed.'
    })
  }

  render() {
    return (
      <div
        className="dropPageBackground"
        style={{ paddingTop: window.innerWidth >= 550 ? '5%' : '20%' }}
      >
        <div className="dropPage">
          <section className="droppedBox">
            <div className="dropZone">
              <Dropzone onDrop={this.onDrop.bind(this)}>
                {({ getRootProps, getInputProps }) => (
                  <section>
                    <div {...getRootProps()}>
                      <input {...getInputProps()} />
                      <p>Drag some files here, or click to select files</p>
                    </div>
                  </section>
                )}
              </Dropzone>
            </div>
            <aside>
              <h4>Files Dropped</h4>
              <ul className="droppedFilesBackground">
                {this.state.files.map(f => (
                  <li className="droppedBox" key={f.name}>
                    {f.name} - {f.size} bytes
                  </li>
                ))}
              </ul>
              <Button
                disabled={this.state.files.length === 0}
                className="right"
                onClick={e => {
                  sendFile(this.state.files[0], this.state.files[0].name)
                  this.toggle()
                }}
              >
                Upload File
              </Button>
              <Modal isOpen={this.state.modal} toggle={this.toggle}>
                <ModalBody>File uploaded - your submission is being processed.</ModalBody>
                <ModalFooter>
                  <Button
                    className="invalidSearchButton"
                    onClick={e => {
                      this.toggle()
                    }}
                  >
                    Return
                  </Button>
                </ModalFooter>
              </Modal>
            </aside>
          </section>
          <hr />
        </div>
      </div>
    )
  }
}

export default Upload
