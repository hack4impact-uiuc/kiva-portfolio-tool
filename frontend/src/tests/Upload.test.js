import React from 'react'
import Upload from '../components/Upload'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

test('Upload renders correctly', () => {
  const component = shallow(<Upload />)

  let tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
