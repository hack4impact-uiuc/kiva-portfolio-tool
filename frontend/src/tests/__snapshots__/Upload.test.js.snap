// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload renders correctly 1`] = `
<div
  className="dropPageBackground"
  style={
    Object {
      "paddingTop": "5%",
    }
  }
>
  <div
    className="dropPage"
  >
    <section
      className="droppedBox"
    >
      <div
        className="dropZone"
      >
        <w
          onDrop={[Function]}
        >
          <Component />
        </w>
      </div>
      <aside>
        <h4>
          Files Dropped
        </h4>
        <ul
          className="droppedFilesBackground"
        />
        <Button
          className="right"
          color="secondary"
          disabled={true}
          onClick={[Function]}
          tag="button"
        >
          Upload File
        </Button>
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={false}
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          role="dialog"
          toggle={[Function]}
          zIndex={1050}
        >
          <ModalBody
            tag="div"
          >
            File uploaded - your submission is being processed.
          </ModalBody>
          <ModalFooter
            tag="div"
          >
            <Button
              className="invalidSearchButton"
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              Return
            </Button>
          </ModalFooter>
        </Modal>
      </aside>
    </section>
    <hr />
  </div>
</div>
`;
